ambari-server/src/main/resources/common-services/ATLAS/0.1.0.2.3/package/scripts/params.py

```diff
@@ -421,3 +429,10 @@ def configs_for_ha(atlas_hosts, metadata_port, is_atlas_ha_enabled, metadata_pro
 # atlas admin login username password
 atlas_admin_username = config['configurations']['atlas-env']['atlas.admin.username']
 atlas_admin_password = config['configurations']['atlas-env']['atlas.admin.password']
+
+# Atlas Passwords Extracted From Credential Store
+if credential_provider:
+    default_credential_shell_lib_path = '/var/lib/ambari-agent/cred/lib'
+    truststore_password = PasswordString(get_password_from_credential_store('truststore.password', credential_provider, os.path.join(default_credential_shell_lib_path, '*'), java64_home, default_credential_shell_lib_path))
```

Comment:
Last parameter for get_password_from_credential_store should be jdk_location, ie jdk_location = config['ambariLevelParams']['jdk_location']

What if the jar is not available in "/var/lib/ambari-agent/cred/lib" ? It can download from jdk_location.

As the alias is hard-coded .... need to document to use this alias, other-wise the password won't be available.

Reply:
https://github.com/apache/ambari/blob/906196b2568f21eb0ad18cc51f71f9c3006627bf/ambari-common/src/main/python/ambari_commons/credential_store_helper.py#L34

The function will autodownload to that folder. The JDK location in other scripts that use this point to this folder. When I was doing my testing, I was getting an error when it tried to get jdk_location = config['ambariLevelParams']['jdk_location']

If it works and it is just my testing environment I can remove the hardcoding and put it at as the jdk_location

Reply:
Please use jdk_location.
Currently the code is downloading and writing it in the same path "/var/lib/ambari-agent/cred/lib".
If in case the jar file is not available then it can download from jdk_location.